lib/cadet_web/controllers/incentives_controller.ex

```diff
@@ -15,6 +15,26 @@ defmodule CadetWeb.IncentivesController do
     )
   end
 
+  def update_progress(conn, %{"uuid" => uuid, "userid" => userid, "progress" => progress}) do
+    progress
+    |> json_to_progress(uuid, userid)
+    |> Goals.upsert_progress(uuid, userid)
+    |> handle_standard_result(conn)
+  end
```

Comment:
This doesn't do any validation on `userid`, so students could update other students' goal progress.

Reply:
Also, missing Swagger documentation for this new endpoint.